examples/platform/linux/AppMain.cpp

```diff
@@ -189,6 +189,12 @@ CHIP_ERROR InitCommissioner()
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ShutdownCommissioner()
+{
+    gCommissioner.Shutdown();
```

Comment:
So looking into this more carefully, this code still doesn't make sense.  The resulting code flow in this file is:

1. Call `InitCommissioner`
2. Call `RunEventLoop` to process events on the current thread.
3. Call `ShutdownCommissioner`

but `InitCommissioner` calls `gCommissioner.ServiceEvents()` which calls `DeviceLayer::PlatformMgr().StartEventLoopTask()`  which starts event processing on a newly-created child thread.

So we end up with two threads racing to process events out of the same queue.  This is explicitly documented as something that is not allowed in the `PlatformManager` API:

> a consumer is allowed to call either StartEventLoopTask or RunEventLoop to process pending work.  Calling both is not allowed: it must be one or the other.

Reply:
Thanks for catching this. I removed the ServiceEvents() call from InitCommissioner.

PlatformMgr should probably throw an exception when its not used as designed rather than appearing to work!

Reply:
> PlatformMgr should probably throw an exception when its not used as designed rather than appearing to work!

Yes, it should....